src/hooks/screen/useChatScreenHook.ts

```diff
@@ -271,23 +271,34 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
   });
 
   const updateChatContinuity = (chatsData: ChatSchema[]) => {
-    const updatedChats = chatsData.map((currentChat, currentIndex) => {
-      const previousChat = chatsData[currentIndex + 1];
-
-      if (previousChat) {
+    const updatedChat = chatsData
+      .map((currentChat, currentIndex) => {
+        const previousChat = chatsData[currentIndex + 1];
+
+        if (previousChat) {
+          if (
+            currentChat?.userId === previousChat?.userId &&
+            (previousChat?.rawJson?.isSystem ||
+              previousChat?.rawJson?.type === 'system' ||
+              previousChat?.rawJson?.message?.type === 'system')
+          ) {
+            currentChat.isContinuous = false;
+          }
+        }
+        // remove current same system message between previous and current chat
         if (
-          currentChat?.userId === previousChat?.userId &&
           (previousChat?.rawJson?.isSystem ||
             previousChat?.rawJson?.type === 'system' ||
-            previousChat?.rawJson?.message?.type === 'system')
+            previousChat?.rawJson?.message?.type === 'system') &&
+          previousChat?.message === currentChat.message
         ) {
-          currentChat.isContinuous = false;
+          return null;
         }
-      }
-      return currentChat;
-    });
+        return currentChat;
+      })
+      .filter((i) => i != null);
 
-    return updatedChats;
+    return updatedChat;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for removing duplicate system messages seems correct. However, the code could be optimized by reducing the number of times we check if a message is a system message. We can store this result in a variable and use it in both conditions. This will make the code cleaner and slightly more performant.

```diff
   const updateChatContinuity = (chatsData: ChatSchema[]) => {
     const updatedChat = chatsData
       .map((currentChat, currentIndex) => {
         const previousChat = chatsData[currentIndex + 1];
 
         if (previousChat) {
+          const isPreviousChatSystem = previousChat?.rawJson?.isSystem ||
+            previousChat?.rawJson?.type === 'system' ||
+            previousChat?.rawJson?.message?.type === 'system';
           if (
             currentChat?.userId === previousChat?.userId &&
-            (previousChat?.rawJson?.isSystem ||
-              previousChat?.rawJson?.type === 'system' ||
-              previousChat?.rawJson?.message?.type === 'system')
+            isPreviousChatSystem
           ) {
             currentChat.isContinuous = false;
           }
         }
         // remove current same system message between previous and current chat
          if (
-            (previousChat?.rawJson?.isSystem ||
-              previousChat?.rawJson?.type === 'system' ||
-             previousChat?.rawJson?.message?.type === 'system') &&
+            isPreviousChatSystem &&
           previousChat?.message === currentChat.message
          ) {
           return null;
          }
         return currentChat;
       })
       .filter((i) => i != null);
  
     return updatedChat;
   };
```
Also, please ensure that `ChatSchema` has all the necessary null checks and default values to prevent potential null pointer exceptions.


<!-- This is an auto-generated comment by OSS CodeRabbit -->